src/sharedwithyoucore.cs

```diff
@@ -0,0 +1,261 @@
+//
+// SharedWithYouCore C# bindings
+//
+// Authors:
+//	Manuel de la Pena Saenz <mandel@microsoft.com>
+//
+// Copyright 2022 Microsoft Corporation All rights reserved.
+//
+
+using System;
+
+using AVFoundation;
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+#if !NET
+using NativeHandle = System.IntPtr;
+#endif
+
+namespace SharedWithYouCore {
+
+	[NoWatch, TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	interface SWAction : NSCopying, NSSecureCoding {
+		[Export ("uuid")]
+		NSUuid Uuid { get; }
+
+		[Export ("complete")]
+		bool Complete { [Bind ("isComplete")] get; }
+
+		[Export ("fulfill")]
+		void Fulfill ();
+
+		[Export ("fail")]
+		void Fail ();
+	}
+
+	interface ISWCollaborationActionHandler {}
+
+	[NoWatch, TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[Protocol]
+	[BaseType (typeof (NSObject))]
```

Comment:
Is this BaseType necessary if this protocol is not a model?

Reply:
Good catch 👍